lib/zzhc/zzhc_port.c

```diff
@@ -0,0 +1,365 @@
+#include <at_cmd.h>
+#include <at_notif.h>
+#include <base64.h>
+#include <secure_services.h>
+#include <cJSON.h>
+#include <cJSON_os.h>
+#include <at_cmd_parser/at_cmd_parser.h>
+#include <settings/settings.h>
+
+#define AT_PARAMS_MAX     10        /** Max. # of AT-params to parse */
+#define ICCID_FILE_ID     0x2B48    /** File ID of stored ICCID */
+#define THREAD_PRIORITY   K_PRIO_PREEMPT(CONFIG_ZZHC_THREAD_PRIO)
+#define STACK_SIZE        CONFIG_ZZHC_STACK_SIZE
+#define FICR_REV_OFFSET   0x134
+
+/** Internal thread stack. */
+K_THREAD_STACK_DEFINE(thread_stack, STACK_SIZE);
+
+/** Cached storage of ICCID. */
+static char ueiccid[ICCID_LEN];
+
+/** Name of settings */
+static char const settings_name[] = "zzhc";
+
+#ifndef LOG_DBG
+#define LOG_DBG(...) \
```

Comment:
Why is this needed, doesn't the logger provide a macro on its own?

Reply:
Some 3rd parties are using this. As mentioned earlier, zzhc.c and zzhc_port.c will be merged into one file near the end of the review of this PR. This dummy stuff will be removed then.

Reply:
In the push on 5th Nov, 2019, the calls to printf() inside such macro have been removed. As said, the library is still being used by 3rd parties, there is a need to leave it temporarily here. As also said, once zzhc.c and zzhc_port.c are merged, such macro will be eliminated.